src/nfa.rs

```diff
@@ -234,17 +235,13 @@ pub struct NfaThreads {
 
 #[derive(Debug)]
 struct Threads {
-    dense: Vec<Thread>,
+    pcs: Vec<usize>,
+    cap_bucket_size: usize,
+    caps: Vec<Option<usize>>,
```

Comment:
What motivated this change in representation? Is there a performance improvement?